src/metabase/query_processor/middleware/fetch_source_query.clj

```diff
@@ -94,42 +96,81 @@
         (log/info (trs "Trimming trailing comment from card with id {0}" card-id))
         trimmed-string))))
 
+(defn sub-cached-field-refs
+  "Change field refs by id into field refs by name."
+  [metadata]
+  (map (fn [m]
+         ;; is it ok to clobber expressions like this? I think so.
```

Comment:
`[:field 1 {:source-field 2}]` and `[:field 1 {:source-field 3}]` are pretty different, same if they have different temporal units or binning options or come from different joins. Not sure what's happening here. Maybe that's ok